stan/math/rev/core/ad_allocator.hpp

```diff
@@ -0,0 +1,48 @@
+#ifndef STAN_MATH_REV_CORE_AD_ALLOCATOR_HPP
+#define STAN_MATH_REV_CORE_AD_ALLOCATOR_HPP
+
+#include <stan/math/rev/core/chainablestack.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * std library compatible allocator that uses AD stack.
+ * @tparam T type of scalar
+ */
+template <typename T>
+struct AD_allocator {
+  using value_type = T;
+
```

Comment:
According to: https://en.cppreference.com/w/cpp/memory/allocator

It looks like there are some other boilerplate types to include.

Reply:
Your link points to docs for `std::allocator`. Actual requirements for allocaters are here: https://en.cppreference.com/w/cpp/named_req/Allocator
They list everything else as optional.